src/praisonai/tests/integration/gateway/test_bind_aware_e2e.py

```
@@ -0,0 +1,246 @@
+        try:
+            # Should start without error (permissive mode)
+            await asyncio.wait_for(self._start_gateway_briefly(gateway), timeout=5.0)
+        except Exception as e:
+            pytest.fail(f"Gateway failed to start on loopback without token: {e}")
+    
+    @pytest.mark.asyncio
+    async def test_external_gateway_without_token_fails(self):
+        """Test that a gateway on external interface fails without a token."""
+        # Create gateway config for external interface
+        config = GatewayConfig(
+            host="0.0.0.0",
```

Ruff (0.15.10)

[error] Possible binding to all interfaces

(S104)